ui/packages/atlasmap/src/UI/dialogs/CustomClassDialog.tsx

```diff
@@ -71,24 +77,28 @@ export const CustomClassDialog: FunctionComponent<ICustomClassDialogProps> = ({
     <ConfirmationDialog
       title={title}
       onCancel={handleOnCancel}
-      onConfirm={value.length > 0 ? handleOnConfirm : undefined}
+      onConfirm={customClassName.length > 0 ? handleOnConfirm : undefined}
       isOpen={isOpen}
     >
       <Form>
         <FormGroup
-          label={"Custom class package name"}
-          fieldId={"name"}
+          label={"Custom class name"}
+          fieldId={"custom-class-name"}
           isRequired={true}
         >
-          <TextInput
-            value={value}
-            onChange={setValue}
-            id={"name"}
+          <FormSelect
+            value={customClassName}
+            aria-label={"Select class name"}
             autoFocus={true}
-            isRequired={true}
-            placeholder={"com.package.class"}
-            data-testid={"class-package-name-text-input"}
-          />
+            onChange={setCustomClassName}
+            data-testid={"custom-class-name-form-select"}
+          >
+            {isOpen &&
+              customClassNames &&
+              customClassNames.map((value, idx) => (
+                <FormSelectOption key={idx} value={value} label={value} />
```

Comment:
I think it would be better to use value as a key here.

Reply:
either way - I used the index as the key.